docs/_posts/2020-02-28-puppetwash.md

```diff
@@ -0,0 +1,283 @@
+    │   ├── bios_vendor
+    │   ├── bios_version
+    │   ├── blockdevice_sda_model
+    │   ├── blockdevice_sda_size
+    │   ├── blockdevice_sda_vendor
+    │   ├── blockdevice_sr0_model
+    │   ├── blockdevice_sr0_size
+    │   ├── blockdevice_sr0_vendor
+    │   ├── blockdevice_sr1_model
+    │   ├── blockdevice_sr1_size
+    │   ├── blockdevice_sr1_vendor
+    │   ├── blockdevice_sr2_model
+    │   ├── blockdevice_sr2_size
+    │   ├── blockdevice_sr2_vendor
+    │   ├── blockdevices
+    │   ├── chassisassettag
+    │   ├── chassistype
+    │   ├── clientcert
+    │   ├── clientnoop
+    │   ├── clientversion
+    │   ├── disks
+    │   ├── dmi
+    │   ├── domain
+    │   ├── facterversion
+    │   ├── filesystems
+    │   ├── fips_enabled
+    │   ├── fqdn
+    │   ├── gid
+    │   ├── hardwareisa
+    │   ├── hardwaremodel
+    │   ├── hostname
+    │   ├── hypervisors
+    │   ├── id
+    │   ├── identity
+    │   ├── interfaces
+    │   ├── ipaddress
+    │   ├── ipaddress_eth0
+    │   ├── ipaddress_lo
+    │   ├── is_virtual
+    │   ├── kernel
+    │   ├── kernelmajversion
+    │   ├── kernelrelease
+    │   ├── kernelversion
+    │   ├── load_averages
+    │   ├── macaddress
+    │   ├── macaddress_eth0
+    │   ├── memory
+    │   ├── memoryfree
+    │   ├── memoryfree_mb
+    │   ├── memorysize
+    │   ├── memorysize_mb
+    │   ├── mountpoints
+    │   ├── mtu_eth0
+    │   ├── mtu_ip6tnl0
+    │   ├── mtu_lo
+    │   ├── mtu_tunl0
+    │   ├── netmask
+    │   ├── netmask_eth0
+    │   ├── netmask_lo
+    │   ├── network
+    │   ├── network_eth0
+    │   ├── network_lo
+    │   ├── networking
+    │   ├── operatingsystem
+    │   ├── operatingsystemmajrelease
+    │   ├── operatingsystemrelease
+    │   ├── os
+    │   ├── osfamily
+    │   ├── partitions
+    │   ├── path
+    │   ├── physicalprocessorcount
+    │   ├── processor0
+    │   ├── processor1
+    │   ├── processor2
+    │   ├── processor3
+    │   ├── processor4
+    │   ├── processor5
+    │   ├── processorcount
+    │   ├── processors
+    │   ├── productname
+    │   ├── puppetversion
+    │   ├── ruby
+    │   ├── rubyplatform
+    │   ├── rubysitedir
+    │   ├── rubyversion
+    │   ├── selinux
+    │   ├── serialnumber
+    │   ├── swapfree
+    │   ├── swapfree_mb
+    │   ├── swapsize
+    │   ├── swapsize_mb
+    │   ├── system_uptime
+    │   ├── timezone
+    │   ├── trusted
+    │   ├── uptime
+    │   ├── uptime_days
+    │   ├── uptime_hours
+    │   ├── uptime_seconds
+    │   ├── uuid
+    │   └── virtual
+    └── reports
+        └── 2020-02-28T00:08:28.495Z
+
+3 directories, 107 files
+```
+
+Pupperware presents the last catalog (`catalog.json`), reports (as JSON), and individual fact values. We can look at fact values with
+```
+wash puppet/pupperware/nodes > cd puppet.test
+wash puppet/pupperware/nodes/puppet.test > cat facts/osfamily
+Debian
+```
+
+Reports also have metadata attached to them that we can filter on
```

Comment:
Could we filter anything in node's metadata?

Reply:
Ah, forgot about that.